Google.PowerShell/Provider/CacheItem.cs

```diff
@@ -17,33 +17,77 @@ public class CacheItem<T>
         private T _value;
         private readonly Func<T> _update;
 
+        /// <summary>
+        /// Get the value after applying the default update function
+        /// if the value is out of date.
+        /// </summary>
         public T Value
         {
             get
             {
-                if (DateTimeOffset.Now > _lastUpdate + _cacheLifetime)
-                {
-                    _value = _update();
-                    _lastUpdate = DateTimeOffset.Now;
-                }
-                return _value;
+                return GetValueWithUpdateFunction(_update);
             }
         }
 
-        public CacheItem(Func<T> update) : this(update, TimeSpan.FromMinutes(1)) { }
+        /// <summary>
+        /// Returns true if the cache is out of date.
+        /// </summary>
+        public bool CacheOutOfDate
+        {
+            get
+            {
+                return DateTimeOffset.Now > _lastUpdate + _cacheLifetime;
+            }
+        }
 
-        public CacheItem(Func<T> update, TimeSpan cacheLifetime)
+        /// <summary>
+        /// Get value after applying an update function to the value
+        /// if the value is out of date.
+        /// </summary>
+        /// <param name="updateFunc"></param>
+        /// <returns></returns>
+        public T GetValueWithUpdateFunction(Func<T> updateFunc)
         {
-            _update = update;
-            _cacheLifetime = cacheLifetime;
+            if (CacheOutOfDate && updateFunc != null)
+            {
+                _value = updateFunc();
+                _lastUpdate = DateTimeOffset.Now;
+            }
+            return _value;
+        }
+
+        /// <summary>
+        /// Get the last stored value without calling the update function.
+        /// </summary>
+        /// <returns></returns>
```

Comment:
Remove the returns tag since the summary pretty much says what is going to be returned.